### PR TITLE
geoclue2 2.1.10 (new formula)

### PR DESCRIPTION
**Pull request by DomT4**
New formula for GeoClue2.

An increasing amount of GNOME stuff is requiring this, boo.

### CODE REVIEW

**Comment by DomT4** [comment]
GNOME *seem* to have decided supporting ` CoreLocation ` on OS X isn't a priority, which is fair enough for them I guess, but they've also decided it'd be amusing to make OS X users get stuck with this shiny new dependency for geolocation sniffing. Frankly, this dependency in GNOME software, where they haven't given people the option to disable it or disabling it breaks useful functionality, stinks :crying_cat_face:.

No test really possible, because it doesn't ship any particularly testable things. We're building here with the same settings as MacPorts do, and I don't think they've got a test for this either.

**Comment by DomT4** [comment]
Everything passed fine aside the expected:

```
==> brew audit geoclue2 --strict					 FAILED
Error: 1 problems in 1 formulae
geoclue2:
 * A `test do` test block should be added
```

File: Library/Formula/geoclue2.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  depends_on "libsoup"
+
+  def install
+    system "./autogen.sh" if build.head?
+    system "./configure", "--prefix=#{prefix}", "--sysconfdir=#{etc}", "--disable-silent-rules",
+                          "--disable-dependency-tracking","--disable-3g-source", "--disable-cdma-source",
+                          "--disable-modem-gps-source", "--localstatedir=#{var}"
+    system "make", "install"
+  end
+end
```
Comment (mikemcquaid) [inline Library/Formula/geoclue2.rb]: Can a test be added to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by tdsmith** [comment]
There's an example CLI client here: http://cgit.freedesktop.org/geoclue/plain/demo/where-am-i.c -- don't suppose that's in the tarball?

**Comment by bfontaine** [comment]
@tdsmith yes it’s in the tarball, located at `./demo/where-am-i.c`.

**Comment by tdsmith** [comment]
Building and/or invoking that sounds like a good test.

**Comment by DomT4** [comment]
Doesn't work. MacPorts can't seem to get it to work either. Open to suggestions.

**Comment by bfontaine** [comment]
It’s built when installing, and the executable is located at:

    $(brew --prefix geoclue2)/libexec/geoclue-2.0/demos/where-am-i

But it doesn’t work:

    ** (where-am-i:59676): CRITICAL **: Failed to connect to GeoClue2 service: Could not connect: No such file or directory

There’s another executable named `geoclue`:

    $(brew --prefix geoclue2)/libexec/geoclue

But here again it doesn’t work:

    (geoclue:59760): Geoclue-CRITICAL **: Failed to acquire name 'org.freedesktop.GeoClue2' on system bus or lost it.

According to [this][1] and [this][2] it must be installed with `--sysconfdir=/etc` and `--prefix=/usr`:

> A couple of things to note here. First, if you don’t use the --prefix=/usr flags, then the dbus files get put in /usr/local/share instead of /usr/share. Second, you have to use root credentials to do the install because you’re installing dubs services.

[1]: http://uxscrutiny.com/2013/10/building-geocode2-from-source/
[2]: http://permalink.gmane.org/gmane.comp.gnome.geoclue/573

**Comment by DomT4** [comment]
Aye, I'm aware. I read through about a year's worth of emails over the last few nights.

IMO, I find this dubious:

> Second, you have to use root credentials to do the install because you’re installing dubs services.

We already support ` d-bus `, which functions fine as non-root. If all Geoclue2 are doing is hopping on ` d-bus `' back, there's no reason why it couldn't or shouldn't work non-root. I consider their replies to that mail thread to be applicable for various Linux distros, but not really applicable to OS X or to Homebrew, MacPorts or Fink.

We are building identically to MacPorts here, and AFAIK, despite MacPorts running under ` sudo `, they have this same fun dance.

The biggest issue here is that (seemingly increasingly) upstream projects that hook into GNOME and want to use geolocation are shoving users into using Geoclue or Geoclue2, despite the fact those pieces of software weren't really designed with OS X in mind, and GNOME software has a pretty wide usage on OS X, despite it obviously not being GNOME's primary intended distro (*to say the least*).

Failing this going in, I'd be tempted to start writing ` ifndef `s into any GNOME software affected by this so it doesn't expect to use Geoclue2, but I'm not sure that's less hacky than including this. I find this particular situation frustrating. If we establish a firm policy against this going in, I'm perfectly content with that situation, to be honest.

**Comment by tdsmith** [comment]
Does software exist that can correctly use geoclue as packaged by this PR?

**Comment by DomT4** [comment]
Give me like, an hour, and I'll test this PR out with ` redshift `, which has the option to use either CoreLocation or Geoclue2. That should let me know with some certainty one way or another whether it works or whether it's a tiny useless placeholder.

**Comment by tdsmith** [comment]
:+1: That'd be great, thanks.

**Comment by DomT4** [comment]
Yup. Geoclue2 does precisely jack on OS X. Closing until a point where it doesn't just bug out on OS X. If any formulae comes up that starts to require it, propose we wrap the blocks in question in ` ifndef `. Theoretically, everyone should provide a configure option with ` --disable-geoclue2 ` or similar, but theoretical isn't already reality, alas.

**Pull request closed without merging**